src/NexusMods.Paths/RelativePath.cs

```diff
@@ -178,7 +178,7 @@ public override int GetHashCode()
     public static implicit operator ReadOnlySpan<char>(RelativePath d) => d.Path;
 
     /// <summary/>
-    public static implicit operator RelativePath(string b) => new(b);
+    public static implicit operator RelativePath(string b) => new(PathHelpers.Sanitize(b, OS));
```

Comment:
I considered doing this in my original PR #345, however, I decided against it. `PathHelpers.Sanitize` is an expensive operation and should only be called on inputs that come from outside the application like parsing CLI arguments, reading a file, and so on.

Automatically sanitizing all relative paths in the implicit converter, and the `.ToRelativePath` extensions method, is essentially disabling the "is sanitized"-assertion completely for relative paths. The assertions exist to enforce correct path handling in our code, and are only useful if we only sanitize our input paths. It's our internally created paths like `static readonly RelativePath Something = "foo\bar"` that need to be checked for correctness.